### PR TITLE
[qtawesome/6.4.0] Add recipe

### PR DESCRIPTION
**Pull request by MartinDelille**
---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by MartinDelille** [comment]
Waiting for #18162

**Review by SSE4 — APPROVED** [review]

**Comment by MartinDelille** [comment]
@prince-chrismc @ericLemanissier Shall I bump qt/5.15.9 dependencies to make this PR works for conan v2 ?

**Comment by ericLemanissier** [comment]
I just did the bump in https://github.com/conan-io/conan-center-index/pull/18617. Let's wait for the automatic merge.

**Comment by MartinDelille** [comment]
Hi @franramirez688 ! I wondered if this PR might be merged one day ?

File: recipes/qtawesome/all/conanfile.py
Comment (franramirez688) [inline recipes/qtawesome/all/conanfile.py:62]: @MartinDelille Sorry for the delay, let's try to add this param to see if it passes for the Conan v2 job.

```suggestion
        self.requires("qt/5.15.9", transitive_headers=True)
```

File: recipes/qtawesome/all/conanfile.py
Comment (franramirez688) [reply recipes/qtawesome/all/conanfile.py:56]: @uilianries It did not work, so I recommend getting back to having only the `transitive_headers=True` param. Any news on the CI part? Next steps to move forward with this PR?

Comment (uilianries) [reply recipes/qtawesome/all/conanfile.py:56]: Not yet. I still can not reproduce that same error. I'll take a look to connect to a CI machine and try to reproduce it. :disappointed:

File: recipes/qtawesome/all/conanfile.py
Comment (franramirez688) [inline recipes/qtawesome/all/conanfile.py:62]: Related to https://github.com/conan-io/conan-center-index/pull/17276/files#r1288202410

```suggestion
        self.requires("qt/5.15.9", transitive_headers=True)
```
It's not necessary anymore.

File: recipes/qtawesome/all/test_v1_package/CMakeLists.txt
Comment (franramirez688) [inline recipes/qtawesome/all/test_v1_package/CMakeLists.txt:1]: I think that we could entirely remove this `test_v1_package`. New recipes are not required to have that folder anymore (see https://github.com/conan-io/conan-center-index/discussions/17406).

Comment (uilianries) [reply recipes/qtawesome/all/test_v1_package/CMakeLists.txt:1]: Good catch!

**Comment by franramirez688** [comment]
@MartinDelille We're almost there! 👏  Sorry for the long time taking to merge this PR, but after those suggestions, I think it'll be ready to be merged 😁

File: recipes/qtawesome/all/test_package/test_package.cpp
Comment (jcar87) [reply recipes/qtawesome/all/test_package/test_package.cpp:11]: Indeed - the instantiation of `QApplication` will require Qt to locate the Cocoa platform plugin. In order to do so properly, we would need a qt.conf file or an environment variable (I believe `QT_PLUGIN_PATH` pointing to the location of the plugins). However this being a test_package being driven through command line only, we want to avoid instantiating a GUI window for the test.  I think removing the instance to `QApplication` is correct, otherwise we would need to set `QT_QPA_PLATFORM` environment variable, perhaps to `minimal` or `offscreen` (in addition to being able to locate the plugin).

Avoiding the instantiation of QApplication should be enough if there's a specific function we can directly test, as proposed by @uilianries.

Comment (uilianries) [reply recipes/qtawesome/all/test_package/test_package.cpp:11]: @jcar87 thank you for your detailed explanation! 💚

**Comment by danimtb** [comment]
I was having a look at this and it seems that there was never a qt package generated for conan v2. I am pushing https://github.com/conan-io/conan-center-index/pull/18376 but it requires a change in our infra to get it running. Thanks for the patience 🙏

**Comment by franramirez688** [comment]
@MartinDelille Sorry for the delay! 😢 

We just relaunched the PR because the binaries missing were there.
Now, the CI is reporting this failure:
```
-- Conan: Including build module from 'C:/J2/w/prod-v2/bsr@2/21740/bfcfc/p/qt60957fae00ec4/p/lib/cmake/Qt5Core/conan_qt_core_extras.cmake'
-- Conan: Including build module from 'C:/J2/w/prod-v2/bsr@2/21740/bfcfc/p/qt60957fae00ec4/p/lib/cmake/Qt5Core/conan_qt_qt5_coreprivate.cmake'
-- Conan: Including build module from 'C:/J2/w/prod-v2/bsr@2/21740/bfcfc/p/qt60957fae00ec4/p/lib/cmake/Qt5Core/Qt5CoreMacros.cmake'
-- Conan: Including build module from 'C:/J2/w/prod-v2/bsr@2/21740/bfcfc/p/qt60957fae00ec4/p/lib/cmake/Qt5Gui/conan_qt_qt5_guiprivate.cmake'
-- Conan: Including build module from 'C:/J2/w/prod-v2/bsr@2/21740/bfcfc/p/qt60957fae00ec4/p/lib/cmake/Qt5Widgets/conan_qt_qt5_widgetsprivate.cmake'
-- Conan: Including build module from 'C:/J2/w/prod-v2/bsr@2/21740/bfcfc/p/qt60957fae00ec4/p/lib/cmake/Qt5Widgets/Qt5WidgetsMacros.cmake'
-- Conan: Including build module from 'C:/J2/w/prod-v2/bsr@2/21740/bfcfc/p/qt60957fae00ec4/p/lib/cmake/Qt5Core/conan_qt_core_extras.cmake'
-- Conan: Including build module from 'C:/J2/w/prod-v2/bsr@2/21740/bfcfc/p/qt60957fae00ec4/p/lib/cmake/Qt5Core/conan_qt_qt5_coreprivate.cmake'
-- Conan: Including build module from 'C:/J2/w/prod-v2/bsr@2/21740/bfcfc/p/qt60957fae00ec4/p/lib/cmake/Qt5Core/Qt5CoreMacros.cmake'
-- Conan: Including build module from 'C:/J2/w/prod-v2/bsr@2/21740/bfcfc/p/qt60957fae00ec4/p/lib/cmake/Qt5Gui/conan_qt_qt5_guiprivate.cmake'
-- Conan: Including build module from 'C:/J2/w/prod-v2/bsr@2/21740/bfcfc/p/qt60957fae00ec4/p/lib/cmake/Qt5Widgets/conan_qt_qt5_widgetsprivate.cmake'
-- Conan: Including build module from 'C:/J2/w/prod-v2/bsr@2/21740/bfcfc/p/qt60957fae00ec4/p/lib/cmake/Qt5Widgets/Qt5WidgetsMacros.cmake'
-- Configuring done
CMake Error: AUTOUIC for target QtAwesome: Test run of "uic" executable "C:/J2/w/prod-v2/bsr@2/21740/bfcfc/p/qt60957fae00ec4/p/lib/cmake/Qt5Core/../../../bin/uic.exe" failed.
C:/J2/w/prod-v2/bsr@2/21740/bfcfc/p/qt60957fae00ec4/p/lib/cmake/Qt5Core/../../../bin/uic.exe -h

Exit code 0xc0000135

CMake Generate step failed.  Build files cannot be regenerated correctly.
```
Do you have any idea about what's happening?

File: recipes/qtawesome/all/conanfile.py
Comment (franramirez688) [inline recipes/qtawesome/all/conanfile.py:62]:
```suggestion
        self.requires("qt/5.15.9", transitive_headers=True, transitive_libs=True)
```

**Comment by franramirez688** [comment]
> 0xc0000135 is the NTSTATUS "DLL Not Found", I think some Qt DLLs are missing in PATH...

Let's see if that resolves the issue

**Comment by MartinDelille** [comment]
Thanks @leha-bot for improving the cmake installation! The ideal would be to wait for a future version that would not require patching. Do you think it could be achievable after your patch ? I could then simple ask the upstream repo owner to release 6.4.3. Not having patches will make future maintenance easier!

File: recipes/qtawesome/all/conanfile.py
Comment (franramirez688) [inline recipes/qtawesome/all/conanfile.py:63]: @danimtb Let's try it without that param given your latest commit https://github.com/conan-io/conan-center-index/pull/17276/commits/5a66e76ff94d96cdb5f82068b9f19835834ea6aa
```suggestion
        self.requires("qt/5.15.9", transitive_headers=True)
```

**Review by franramirez688 — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Comment by MartinDelille** [comment]
I missed that! Thank for helping me merging this! 🎉

**Pull request merged**
